Disable memset warning that happens when the fill-value of memset is a non-zero
constant and the size parameter is zero.

This warning is triggered when certain functions containing memset calls are
inlined, example:

https://bugs.webkit.org/show_bug.cgi?id=78513

Warnings are treated as errors in the Chromium build.


--- string/bits/string3.h	2012-02-17 14:32:47.830600426 -0800
+++ string/bits/string3.h	2012-02-17 14:32:47.830600426 -0800
@@ -20,9 +20,6 @@
 # error "Never use <bits/string3.h> directly; include <string.h> instead."
 #endif
 
-__warndecl (__warn_memset_zero_len,
-	    "memset used with constant zero length parameter; this could be due to transposed parameters");
-
 #ifndef __cplusplus
 /* XXX This is temporarily.  We should not redefine any of the symbols
    and instead integrate the error checking into the original
@@ -79,7 +76,6 @@
   if (__builtin_constant_p (__len) && __len == 0
       && (!__builtin_constant_p (__ch) || __ch != 0))
     {
-      __warn_memset_zero_len ();
       return __dest;
     }
   return __builtin___memset_chk (__dest, __ch, __len, __bos0 (__dest));
